jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/preprocessor/UserAndGroupsTasksPreprocessor.java

```diff
@@ -105,4 +109,27 @@ private void addUserAndGroupsFromIdentityProvider(List<Comparable> orgEntities,
         orgEntities.addAll(Optional.ofNullable(userGroupCallback.getGroupsForUser(userId)).orElse(new ArrayList<>()));
         orgEntities.add(userId);
     }
+
+    private void removeCoreFunctionFilter(List<ColumnFilter> filters, ColumnFilter removedFilter) {
+        if (filters.remove(removedFilter)) {
+            return;
+        }
+
+        filters.stream()
+               .filter(e -> e instanceof LogicalExprFilter)
+               .map(e -> (LogicalExprFilter) e)
+               .forEach(filter -> removeCoreFunctionFilter(filter.getLogicalTerms(), removedFilter));
+
+    }
+    private List<CoreFunctionFilter> extractCoreFunctionFilter(List<ColumnFilter> filters) {
+        List<CoreFunctionFilter> list = new ArrayList<>();
+        for (ColumnFilter filter : filters) {
+            if (filter instanceof CoreFunctionFilter) {
+                list.add((CoreFunctionFilter) filter);
+            } else if (filter instanceof LogicalExprFilter) {
+                list.addAll(extractCoreFunctionFilter(((LogicalExprFilter) filter).getLogicalTerms()));
+            }
```

Comment:
What happens for other instances of filters that are not `CoreFunctionFilter` neither `LogicalExprFilter`?

Reply:
the other won't apply here or it should not be here.